lib/neovim-plugin.nix

```diff
@@ -161,8 +162,48 @@
                   # Add the plugin setup code `require('foo').setup(...)` to the lua configuration
                   (lib.optionalAttrs callSetup { ${namespace}.${name}.luaConfig.content = setupCode; })
 
-                  # Write the lua configuration `luaConfig.content` to the config file
-                  (setLuaConfig cfg.luaConfig.content)
+                  # Write the lua configuration `luaConfig.content` to the config file when lazy loading is not enabled
+                  (lib.mkIf (!cfg.lazyLoad.enable) (setLuaConfig cfg.luaConfig.content))
+
+                  # When lazy loading is enabled for this plugin, route its configuration to the enabled provider
+                  (lib.mkIf cfg.lazyLoad.enable {
+                    assertions = [
+                      {
+                        assertion = (isColorscheme && colorscheme != null) || cfg.lazyLoad.settings != { };
+                        message = "You have enabled lazy loading for ${originalName} but have not provided any configuration.";
+                      }
+                    ];
+
+                    warnings = lib.optional (!config.plugins.lz-n.enable) ''
+                      You have enabled lazy loading for ${originalName} but have not enabled any lazy loading plugins.
+                      We will not insert the lua for this plugin and you must handle loading it yourself.  
+
+                      Currently supported lazy providers:
+                        - lz-n
+                    '';
```

Comment:
This means that if someone wants to use an unsupported lazy loader they will have warnings for each of their plugins right?

In another PR we should consider adding a global option that will allow to silence the warnings, something like `useUnsupportedLazyLoader`